### PR TITLE
Add rules_shell dep to rules_bazel_integration_test

### PR DESCRIPTION
**Pull request by ted-xie**
native.{sh_test,sh_library} no longer exist in Bazel HEAD.

### CODE REVIEW

**Review by katre — APPROVED** [review]

**Comment by katre** [comment]
Looks like the MODULE.bazel.lock file is stale. Upload a new version and I'll re-approve

**Comment by ted-xie** [comment]
@katre I updated the MODULE lockfile, I think -- I did `rm MODULE.bazel.lock; bazelisk build ... --lockfile_mode=update`.

**Review by katre — APPROVED** [review]

**Comment by katre** [comment]
Now the error is generate the stardoc:
```
com.google.devtools.build.skydoc.SkydocMain$StarlarkEvaluationException: File /home/runner/.cache/bazel/_bazel_runner/331b13f68cad845627c8e44b927d4117/sandbox/linux-sandbox/35/execroot/_main/bazel-out/k8-opt-exec-ST-d57f47055a04/bin/doc/rules_and_macros_overview_stardoc.runfiles/_main/bazel_integration_test/private/bazel_integration_test.bzl imported '@rules_shell//shell:sh_test.bzl', yet /home/runner/.cache/bazel/_bazel_runner/331b13f68cad845627c8e44b927d4117/sandbox/linux-sandbox/35/execroot/_main/bazel-out/k8-opt-exec-ST-d57f47055a04/bin/doc/rules_and_macros_overview_stardoc.runfiles/rules_shell/shell/sh_test.bzl was not found.
```

I'm not sure how to resolve this, I assume you need to add the rules_shell dependency somewhere else but I don't know how the docs generation works.

I can take a look with you next week if that helps.

**Comment by ted-xie** [comment]
Turns out I just needed to add `@rules_shell//shell:rules_bzl` to the deps for bazel_integration_test.

**Comment by cgrindel** [comment]
@ted-xie Do `rules_sh` work with Bazel 7.x?

**Comment by ted-xie** [comment]
@cgrindel Yes, we're currently using rules_shell + bazel 7.4 in rules_android.

**Comment by cgrindel** [comment]
@ted-xie Can you just delete `MODULE.bazel.lock`? We really do not need this in a ruleset repo.

**Review by katre — APPROVED** [review]

**Comment by ted-xie** [comment]
Happy to delete MODULE.bazel.lock in a follow-up PR, or in a follow-up commit in this current PR if the presubmit fails again.

**Comment by cgrindel** [comment]
To make sure that everything is up-to-date, run `bazel run //:tidy`.

**Comment by katre** [comment]
You need to rebase onto the latest main anyway.

**Comment by ted-xie** [comment]
Rebased to latest main, deleted the lockfile, and ran tidy.

**Comment by cgrindel** [comment]
Once we get this merged, I think that we should run release unless there are other PRs that we want to wait on. 

@katre @ted-xie Does that sound right to you?

File: .bazelrc
Comment (katre) [inline .bazelrc:6]: Why did this change? Something from the tidy script?

**Review by katre — APPROVED** [review]

File: .bazelrc
Comment (ted-xie) [reply .bazelrc:6]: Yeah, this was from tidy.

**Comment by katre** [comment]
Cutting a release sounds good.

File: .bazelrc
Comment (cgrindel) [reply .bazelrc:6]: There is a task in `//:tidy` that ensures that list is up-to-date. Perhaps, it was out-of-sync.

**Comment by katre** [comment]
Looks like the bazelrc change is causing problems, so go ahead and revert that.

File: .bazelrc
Comment (cgrindel) [reply .bazelrc:6]: On second look, that does not look right. Maybe something is wrong with the script. 👀

**Comment by cgrindel** [comment]
Give me a second, I'll pull it down and check.

**Comment by ted-xie** [comment]
It looks like the diff that got introduced was basically this:

Before:
```
custom_test_runner/Sources/CustomTestRunner,examples
```
After:
```
custom_test_runner/integration_tests,examples
```

I'm not familiar with this repo and its history, so I'll leave that to @cgrindel to ponder, and in the mean time I'll revert my change to the bazelrc.

**Comment by cgrindel** [comment]
@ted-xie Don't make any changes. I'll just push to your branch.

**Comment by cgrindel** [comment]
Nevermind. 🙂

**Comment by katre** [comment]
I'm heading out to the airport soon so I'll leave the rest of the review to @cgrindel.

**Comment by ted-xie** [comment]
I deleted the commit when I un-reverted the bazelrc change (whew!). If you have a more complete solution, feel free to push to my branch :)

**Review by cgrindel — APPROVED** [review]

**Pull request merged**
